### PR TITLE
ZEPPELIN-3108. Support Spark 2.3

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
Spark 2.3 remove `JobProgressListener` which cause zeppelin unable to run spark 2.3.  
This PR try to make Zeppelin support spark 2.3 via using `sc.statusTracker`, see `JobProgressUtil.scala`

### What type of PR is it?
[Improvement ]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3108

### How should this be tested?
* Verified manually. 

### Screenshots (if appropriate)
screen shot 2018-01-30 at 9 45 01 pm

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: spark-dependencies/pom.xml [hunk 1]
```diff
@@ -565,6 +565,15 @@
       </properties>
     </profile>
 
+    <profile>
```
Comment (felixcheung) [inline spark-dependencies/pom.xml]: can you update to 2.1.2 and 2.2.1 above?

Comment (zjffdu) [reply spark-dependencies/pom.xml]: Fixed

Comment (zjffdu) [reply spark-dependencies/pom.xml]: @felixcheung , looks like spark-2.2.1 is not uploaded in http://d3kbcqa49mib13.cloudfront.net/spark-2.2.1.tgz which is used for zeppelin to download it for integration test. Can you help check what's missing in spark side ?

Comment (zjffdu) [reply spark-dependencies/pom.xml]: Same issue for spark-2.1.2.  http://d3kbcqa49mib13.cloudfront.net/spark-2.1.2.tgz

**Comment by felixcheung** [comment]
The s3 is no longer being used. We need to move to apache mirrors as per ASF policy..

**Comment by zjffdu** [comment]
@felixcheung , download url is fixed, please help review again.

**Comment by felixcheung** [comment]
Hmm we cant fix to one or two mirrors either, i think they tend to disappear - please see how other projects handles this

**Comment by zjffdu** [comment]
Fixed. Now using apache dist url

**Comment by felixcheung** [comment]
Ah sorry.
Please see
http://www.apache.org/dev/release-publishing.html#distribution

And specifically
http://www.apache.org/dev/release-download-pages.html#links

**Comment by zjffdu** [comment]
@felixcheung IIUC, these are for release download page, for external project, we could use the dist link.

**Comment by felixcheung** [comment]
well... that's not my understand and other implementations are pulling from the download script and then to the referred mirror, eg.

https://github.com/apache/spark/blob/master/build/mvn#L81
https://github.com/apache/spark/blob/master/sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala#L64
https://github.com/apache/spark/blob/master/R/pkg/R/install.R#L213
https://github.com/apache/spark/blob/master/external/docker/spark-mesos/Dockerfile#L27
https://github.com/graphframes/graphframes/commit/a1427217a2458ae89dde038f0ce79eec7913a7d6
https://github.com/Homebrew/homebrew-core/blob/master/Formula/apache-spark.rb

**Comment by zjffdu** [comment]
Thanks @felixcheung , These examples seems to fetch the download url dynamically, I don't think maven download plugin support this. If you don't mind, I will leave it as follow up PR to improve. As I want to include this PR in 0.8 which I plan to release soon.

**Comment by felixcheung** [comment]
I don't feel strongly about it, but fetching the url from the download script (.cgi url) is actually fairly simple to do, but I admit may not be what the maven download plugin support.

I don't remember exactly what these downloads are for ... can some of these be artifacts on maven central repo? that might be easier.

in any case, I'm only worry about push back from the ASF release process of downloading directly from dist.apache.org

**Comment by zjffdu** [comment]
These download are for spark interpreter integration test. So spark distribution is needed, only spark artifact is not sufficient.

**Comment by zjffdu** [comment]
ping @felixcheung

**Comment by felixcheung** [comment]
Well, my question on the download link
https://github.com/apache/zeppelin/pull/2750#issuecomment-363663111

**Pull request closed without merging**